scim2-sdk-common/src/test/java/com/unboundid/scim2/common/SearchRequestTestCase.java

```diff
@@ -32,83 +32,160 @@
 
 package com.unboundid.scim2.common;
 
-import com.unboundid.scim2.common.exceptions.ScimException;
+import com.unboundid.scim2.common.filters.Filter;
 import com.unboundid.scim2.common.messages.SearchRequest;
 import com.unboundid.scim2.common.messages.SortOrder;
+import com.unboundid.scim2.common.types.UserResource;
 import com.unboundid.scim2.common.utils.JsonUtils;
-import com.unboundid.scim2.common.utils.StaticUtils;
 import org.testng.annotations.Test;
 
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertNull;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * Test for search requests.
  */
 public class SearchRequestTestCase
 {
   /**
-   * Test search request.
-   *
-   * @throws java.io.IOException If an error occurs.
-   * @throws ScimException If an error occurs.
+   * Verifies that serialization and deserialization of search request objects
+   * result in an expected structure.
    */
   @Test
-  public void testSearchRequest() throws Exception
+  public void testSerialization() throws Exception
   {
-    // Test case insensitivity
-    SearchRequest searchRequest = JsonUtils.getObjectReader().
```

Comment:
I added my new test case and realized it also encompassed some behavior tested by this existing test. I removed the old test since the newer test matches other serialization tests better.